sdks/java/io/rrio/src/main/java/org/apache/beam/io/rrio/Caller.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.io.rrio;
+
+import java.io.Serializable;
+
+/** {@link Caller} interfaces user custom code intended for API calls. */
+public interface Caller<RequestT, ResponseT> extends Serializable {
```

Comment:
This is a _very_ similar interface to SerializableFunction. Should we consider using that instead, perhaps?

Reply:
The problem with a `SerializeableFunction` is we can't enforce a `throws MySpecificException`.  A `ProcessFunction` is the next candidate but it is just a generic `throws Exception`. I wanted something very specific to signal the kind of error.

Reply:
@johnjcasey this thread shows outdated because of the old package name.

Reply:
sgtm